### PR TITLE
docker: Automate deletion of docker static Jenkins nodes

### PR DESCRIPTION
**Pull request by Haroon-Khel**
- [x] commit message has one of the [standard prefixes](https://github.com/adoptium/infrastructure/blob/master/CONTRIBUTING.md#commit-messages)
- [ ] [faq.md](https://github.com/adoptium/infrastructure/blob/master/FAQ.md) updated if appropriate
- [x] other documentation is changed or added (if applicable)
- [ ] playbook changes run through [VPC](https://ci.adoptium.net/view/Tooling/job/VagrantPlaybookCheck/) or [QPC](https://ci.adoptium.net/view/Tooling/job/QEMUPlaybookCheck/) (if you have access)
- [x] VPC/QPC not applicable for this PR
- [ ] for inventory.yml changes, bastillion/nagios/jenkins updated accordingly

ref https://github.com/adoptium/infrastructure/issues/3506

The python script does most of the work; it deletes the node in Jenkins (if the node is idle). 
Ansible then looks up the machines names and ports in [DockerInventory.json](https://github.com/adoptium/infrastructure/blob/master/ansible/DockerInventory.json) and removes the associated container on the dockerhost machine.
The python script uses a jenkins username and api token, so these need to be added to the secrets repo.

### CODE REVIEW

**Comment by sxa** [comment]
Does this mean you got around the problems you were having with calling in from python?

File: ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/remove_container/tasks/main.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    executable: python3
+  delegate_to: localhost
+  no_log: true
+  register: docker_ports
+
+- name: Set docker_ports variable
+  set_fact:
+    ports_list: "{{ docker_ports.stdout_lines[0].split(' ') | list }}"
+
+# Iterate over each machine, removes conatiner on dockerhost machine
```
Comment (sxa) [inline ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/remove_container/tasks/main.yml:19]:
```suggestion
# Iterate over each machine, removes container on dockerhost machine
```

File: ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/remove_container/tasks/main.yml [hunk 2]
```diff
@@ -0,0 +1,24 @@
+---
+# The deleteNodes.py script deletes the nodes in Jenkins
+# Ansible then removes the docker containers on the dockerhost via the delete_container role
+# If the node is not idle in Jenkins, the node is not deleted
+
+# no_log hides the api_token from the output
+- name: Run deleteNodes script
+  script: scripts/deleteNodes.py {{ jenkins_username }} {{ jenkins_api_token }} {{ delete_nodes }}
```
Comment (sxa) [inline ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/remove_container/tasks/main.yml:8]: I think I'm missing a little bit of higher level context here - where is `{{ delete_nodes }}` coming from i.e. how is it selecting what to delete? This seems to be the only reference to that variable in this PR so it possibly needs a bit of clarification

**Review by sxa — COMMENTED** [review]
This should have an update to the doc at https://github.com/adoptium/infrastructure/blob/master/ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/DockerStatic/README.md to cover the use of this new functionality :-)

File: ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/remove_container/tasks/main.yml (hunk 2 above)
Comment (Haroon-Khel) [reply ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/remove_container/tasks/main.yml:8]: Im working on updating the docs to explain all of this. {{ delete_nodes }} will be passed to the playbook as an ansible extra variable

**Comment by Haroon-Khel** [comment]
Docs update here https://github.com/adoptium/infrastructure/pull/3562

**Comment by sxa** [comment]
> Docs update here #3562

Can the doc change be cherry-picked across to this PR? It would be nicer to have the doc update and the code change in one.

**Comment by Haroon-Khel** [comment]
> Can the doc change be cherry-picked across to this PR? It would be nicer to have the doc update and the code change in one.

Done

File: ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/remove_container/scripts/deleteNodes.py [hunk 3]
```diff
@@ -0,0 +1,59 @@
+import jenkins
+import sys
+import json
+import requests
+
+def createServer(username, password):
+    url = "http://ci.adoptium.net:80"
```
Comment (sxa) [inline ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/remove_container/scripts/deleteNodes.py:7]: This should be using https I'd have thought - we can't have credentials being passed across unencrypted. We're using https for the `doDelete` operation later on in this file.

Comment (Haroon-Khel) [reply ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/remove_container/scripts/deleteNodes.py]: Good spot

**Review by sxa — APPROVED** [review]
LGTM now - thanks :-)

**Review by steelhead31 — APPROVED** [review]
LGTM, semgrep warning is related to a sample command in docs, and can be ignored.

**Pull request merged**